### PR TITLE
Add source metadata to reuses

### PR DESCRIPTION
**Pull request by clementbiron**
This changeset add validation for `source` metadata for reuses.

### CODE REVIEW

**Comment by clementbiron** [comment]
We need to handle what to do with `source` metadata from `Revenu de base`, `NSW Legislation Explorer`, `Mes aides` beacause I didn't know how to fil them.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,6 +26,9 @@ author:
   name: OpenFisca  # 120 characters max
   contact: contact@openfisca.org  # this email address will only be used by the Core team for questions related to the reuse
   link: https://openfisca.org  # optional URL of the author organisation
+source:
+  repository: https://github.com/openfisca/legislation-explorer # optional URL
+  license: AGPL-3.0-only # optional SPDX license identifier, see https://spdx.org/licenses/
```
Comment (MattiSG) [inline CONTRIBUTING.md:31]: I prefer we don't insist on the “optional” part and instead focus on the “proprietary” option below as an alternative, but consider it a given that contributors should provide their source 🙂 

```suggestion
  repository: https://github.com/openfisca/legislation-explorer  # URL of the source code
  license: AGPL-3.0-only  # SPDX license identifier, see https://spdx.org/licenses/
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -36,6 +39,14 @@ If a reuse becomes disused, it should still be referenced for historical purpose
 until: 2022-12-31  # optional, YYYY-MM-DD format
 ```
 
+#### Proprietary source
+
+If the source is proprietary, please indicate it as a comment with the following format:
```
Comment (MattiSG) [inline CONTRIBUTING.md:44]:
```suggestion
If the source code of your reuse is not open, do not fill in the `source` parameter and indicate it as a comment with the following format:
```

**Review by MattiSG — APPROVED** [review]
Thanks! I am not so happy with the current validation system that prevents us from expressing this “either proprietary or both repository and license must be defined” condition, but it is already very good this way 👍 🚀

**Comment by MattiSG** [comment]
> what to do with source metadata from:

> Revenu de base

I don't see any reference to the source that enabled the article. I thus believe it is missing.

> NSW Legislation Explorer

That reuse should be removed anyway. I believe Legislation Explorer is part of Core and should not be presented for every jurisdiction if it is the default one (as opposed to Rapu Tire, for example).

> Mes aides

Let's just add https://github.com/sgmap/mes-aides-ui (which now redirects to https://github.com/betagouv/aides-jeunes, but as history I believe the old one is better).

**Pull request merged**